codec/packer.go

```diff
@@ -104,21 +120,24 @@ func (p *Packer) PackPublicKey(src crypto.PublicKey) {
 	p.p.PackFixedBytes(src[:])
 }
 
+// UnpackPublicKey crypto.PublicKey into [dest].
 func (p *Packer) UnpackPublicKey(required bool, dest *crypto.PublicKey) {
 	copy((*dest)[:], p.p.UnpackFixedBytes(crypto.PublicKeyLen))
 	if required && *dest == crypto.EmptyPublicKey {
-		p.p.Errs.Add(fmt.Errorf("%w: PublicKey field is not populated", ErrFieldNotPopulated))
+		p.addErr(fmt.Errorf("%w: PublicKey field is not populated", ErrFieldNotPopulated))
 	}
 }
 
 func (p *Packer) PackSignature(src crypto.Signature) {
 	p.p.PackFixedBytes(src[:])
 }
 
+// UnpackPublicKey crypto.Signature into [dest].
+// TODO: should add required param?
 func (p *Packer) UnpackSignature(dest *crypto.Signature) {
```

Comment:
some functions do not have required, should we be consistent and add required to all?